File: Difficulty: Easy/Reverse a Doubly Linked List/reverse-a-doubly-linked-list.cpp
/*
class DLLNode {
  public:
    int data;
    DLLNode *next;
    DLLNode *prev;

    DLLNode(int val) {
        data = val;
        this->next = NULL;
        this->prev = NULL;
    }
};
*/
class Solution {
  public:
    // Function to reverse a doubly linked list
    DLLNode* reverseDLL(DLLNode* head) {
        // Your code here
        DLLNode *temp=head;
        DLLNode *newNode=NULL;
        while(temp){
            newNode=temp;
            DLLNode *front=temp->next;
            temp->next=temp->prev;
            temp->prev=front;
            temp=front;
        }
        return newNode;
        
    }
};

File: Difficulty: Easy/Reverse a Doubly Linked List/reverse-a-doubly-linked-list.java
/*
class Node {
    int data;
    Node next;
    Node prev;

    Node(int data) {
        this.data = data;
        this.next = null;
        this.prev = null;
    }
}
*/
class Solution {
   public Node reverse(Node head) {
        Node temp = head;
        Node back = null;
        if (temp == null)
            return temp;
        while (temp != null) {
            back = temp;
            Node front = temp.next;
            temp.next = temp.prev;
            temp.prev = front;
            temp = front;
        }
        return back;

        // code here

    }
}